src/core/render/index.js

```diff
@@ -48,22 +46,35 @@ function renderMain(html) {
   }
 
   this._renderTo('.markdown-section', html);
+
   // Render sidebar with the TOC
   !this.config.loadSidebar && this._renderSidebar();
 
-  // Execute script
-  if (
-    this.config.executeScript !== false &&
-    typeof window.Vue !== 'undefined' &&
-    !executeScript()
-  ) {
-    setTimeout(_ => {
-      const vueVM = window.__EXECUTE_RESULT__;
-      vueVM && vueVM.$destroy && vueVM.$destroy();
-      window.__EXECUTE_RESULT__ = new window.Vue().$mount('#main');
-    }, 0);
-  } else {
-    this.config.executeScript && executeScript();
+  // Execute markdown <script>
+  if (this.config.executeScript || 'Vue' in window) {
+    executeScript();
+  }
+
+  // Handle Vue content not handled by markdown <script>
+  if ('Vue' in window) {
+    const mainElm = document.querySelector('#main') || {};
+    const childElms = mainElm.children || [];
+
+    for (let i = 0, len = childElms.length; i < len; i++) {
+      const elm = childElms[i];
+      const isValid = ['SCRIPT'].indexOf(elm.tagName) === -1;
+      const isVue = Boolean(elm.__vue__ && elm.__vue__._isVue);
+
+      if (isValid && !isVue) {
+        new window.Vue({
+          mounted: function() {
+            if (this.$children.length === 0) {
+              this.$destroy;
```

Comment:
Should this be a method call, `this.$destroy();`?

How does this code work? I mean, assuming the method-call fix, what's the goal?

Reply:
@anikethsaha @Koooooo-7 @sy-records Let's be careful not to approve PRs without fully checking the code. Adding test code will veriy that things work as expected.

Reply:
The old code was destroying the old instance from the previous page using the `__EXECUTE_RESULT__` ref. How does this new implementation (assuming method-call fix) do it?

Reply:
> Should this be a method call, this.$destroy();?

Yes. Each view instance contains its own `destroy()` method.

The `this.$children.length` checks for the existence of child Vue components (not child DOM nodes). It's harmless (because there will never be child components for docsify auto-mounted/destroyed Vue instances) but unnecessary. I'll remove it.

> How does this code work? I mean, assuming the method-call fix, what's the goal?

This is an alternative approach to docsify's "if we detect Vue, we'll automatically process Vue content without you having the add a <script> tag and manually call `new Vue()` in your markdown files" behavior. 

Previously, docsify handled this by creating a Vue instance out of the `#main` element on each page load. The problem with this approach is that it makes it impossible to render dynamic Vue content because either 1) dynamic Vue content is handled first by processing markdown `<script>`, then docsify tramples it when it mounts `#main` as a Vue instance, or 2) docsify mounts `#main` as a Vue instance first, which breaks Vue because docsify is calling `new Vue()` without the properties (`data`, `methods`, etc.) needed to initialize the dynamic Vue content contained in the markdown. It's a chicken-or-the-egg issue.

This new approach allows docisfy to continue automatically processing Vue content while allowing dynamic Vue content to exist. Instead of mounting the `#main` element as a Vue instance, docsify iterates over the children of `#main`, mounts each element that is not already a Vue instance, then immediately destroys the instance because rending is complete. There is no need to retain Vue instance references (like `__EXECUTE_RESULT__`) because these auto-generated Vue instances have no state, no methods, no hooks, etc. For this type of content, Vue is just acting as a very basic template engine. See the [Basic Rendering](https://docsify-preview-i1zwgvmdb.vercel.app/#/vue?id=basic-rendering) section of the updated docs for examples.

Reply:
@trusktr: `this.$destroy` updated to `this.$destroy()` here: https://github.com/docsifyjs/docsify/pull/1271/commits/b135f8417d9b27b58a545e9f792e70d7b319cc15#diff-250dfe7435d59fa33e80765c47190dd5